pandas/core/reshape/concat.py

```diff
@@ -497,38 +497,36 @@ def _clean_keys_and_objs(
         if isinstance(objs, abc.Mapping):
             if keys is None:
                 keys = list(objs.keys())
-            objs = [objs[k] for k in keys]
+            objs_list = [objs[k] for k in keys]
         else:
-            objs = list(objs)
+            objs_list = list(objs)
 
-        if len(objs) == 0:
+        if len(objs_list) == 0:
             raise ValueError("No objects to concatenate")
 
         if keys is None:
-            objs = list(com.not_none(*objs))
+            objs_list = list(com.not_none(*objs_list))
         else:
             # GH#1649
             clean_keys = []
             clean_objs = []
             if is_iterator(keys):
                 keys = list(keys)
-            if is_iterator(objs):
-                objs = list(objs)
```

Comment:
This if should not be needed as `objs` was always a list at that point - let's see whether the tests agree.